src/install/repository.zig

```diff
@@ -152,6 +154,14 @@ pub const Repository = extern struct {
         argv: []const string,
     ) !string {
         var std_map = try env.map.stdEnvMap(allocator);
+        // Run git clone in non-interactive mode and
+        // error if ssh or https asks for a password
+        // so that we can fallback to ssh mode
+        try std_map.unsafe_map.put("GIT_TERMINAL_PROMPT", "0");
```

Comment:
I think we might want to copy the behavior here https://github.com/npm/cli/blob/22731831e22011e32fa0ca12178e242c2ee2b33d/node_modules/%40npmcli/git/lib/opts.js#L1-L6. Looks like we should use the existing values for `GIT_ASKPASS` and `GIT_SSH_COMMAND`, and if they don't exist use the default values `'echo'` and `'ssh -oStrictHostKeyChecking=accept-new'`

Reply:
Also these keys will need to be placed in `env` before calling `stdEnvMap()`. It might be easiest to use `putAllocKeyAndValue`. And the original values need to be restored, I would put that in the `defer { env.map.deinit }` block

Reply:
Thanks! Good catch for the Host Checking, hadn't thought about that. In the case that the user's system has the existing values to ask for password, we would need to suspend the stdout output of the installer so that the user has a chance to enter the credentials. Otherwise Bun prints over the prompt, do you have advice on how to achieve that?

Reply:
@dylan-conway for the clone, in NPM for Windows they also use the `core.longpaths=true` cli option. Because on Windows, git is limited to 260 chars for paths to install to. Shall Bun install do the same?

Reply:
Good idea, we should include it as an option in `checkout/clone` like `git clone -c core.longpaths=true <repo-url>` to avoid spawning another process.

> Otherwise Bun prints over the prompt, do you have advice on how to achieve that?

I'm not sure what we can do, for now I guess we hope the vscode prompt is chosen over terminal. Need to check how npm/yarn/pnpm handles this